File: mathGeometry/149_max_points_on_a_line.go
func maxPoints(points [][]int) int {
    res := 1
    for i := 0; i < len(points); i++ {
        count := make(map[float64]int)
        p1 := points[i]
        for j := i + 1; j < len(points); j++ {
            p2 := points[j]
            slope := math.Inf(1)
            if p1[0] != p2[0] {
                slope = float64(p2[1] - p1[1]) / float64(p2[0] - p1[0])
            }
            count[slope]++
            res = int(math.Max(float64(res), float64(count[slope] + 1)))
        }
    }
    return res
}

File: mathGeometry/149_max_points_on_a_line.py
class Solution:
    # O(n^2) time | O(n) space - where n is points.length
    def maxPoints(self, points: List[List[int]]) -> int:
        n = len(points)
        if n == 1:
            return 1
        result = 1
        for i in range(n):
            count = collections.defaultdict(int)
            p1 = points[i]
            for j in range(i + 1, n):
                p2, slope = points[j], float('inf')
                if p2[0] != p1[0]:
                    slope = (p2[1] - p1[1]) / (p2[0] - p1[0])
                count[slope] += 1
                result = max(result, count[slope] + 1)
        return result

